test/module/irohad/ordering/on_demand_ordering_gate_test.cpp

```diff
@@ -178,3 +180,51 @@ TEST_F(OnDemandOrderingGateTest, EmptyEventNoProposal) {
 
   ASSERT_TRUE(gate_wrapper.validate());
 }
+
+/**
+ * @given initialized ordering gate
+ * @when batch1 is propagated to ordering gate while cache contains batch2
+ * @then all transactions from batch1 and batch2 are propagated to network
```

Comment:
Batches should be resent on block or empty events instead of `propagateBatch` call.